### PR TITLE
Challenge

### PR DESCRIPTION
**Pull request by DeVuDeveloper**
**Refactoring and Setup:**

- Added essential gems for enhanced functionality: `simple_form`, `turbo-rails`, `stimulus-rails`, `redis`, and `sidekiq`.
- Significantly improved page rendering by leveraging turbo frames and turbo streams.
- Implemented form validation to provide a smoother user experience.
- Conducted comprehensive tests prior to actual code implementation.
- Refactored the codebase and introduced view components for better maintainability and readability.
- Added tests before starting the problem-solving process, ensuring that new features and changes are thoroughly validated.

### **Issue 1: "Warning: Each child in an array or iterator should have a unique 'key' prop."**

**Description:** The React application is showing a warning "Each child in an array or iterator should have a unique 'key' prop." This warning occurs when rendering a list of elements, and a unique `key` attribute is not provided for each element.

**Solution:**

The issue was resolved by updating the `OrderTable` component to provide a unique `key` attribute for each `OrderRow` component that is rendered within the `.map()` function. The `key` attribute is set to the `id` of each order to ensure uniqueness.

```jsx
import React from 'react';
import OrderRow from './OrderRow';

const OrderTable = ({ orders }) => (
  <table className="table orders-table">
    <thead>
      <tr>
        <th>Order #</th>
        <th>Ordered at</th>
        <th>Pick up at</th>
        <th>Customer Name</th>
        <th>Item</th>
        <th>Qty</th>
        <th>Status</th>
        <th>Actions</th>
      </tr>
    </thead>
    <tbody>
      {orders.map((order) => (
        <OrderRow key={order.id} order={order} /> // Added a unique 'key' attribute with the id
      ))}
    </tbody>
  </table>
);

export default OrderTable;
```
**Refactored Code and Folder Structure**

I have refactored the code and updated the folder structure to improve organization and readability. The changes include:

1. Created a new "hooks" folder to store custom hooks, and placed the `useFetchOrders` hook inside it.

2. Moved the `OrderRow` and `OrderTable` components into separate files inside the "components/orders" folder.

3. Utilized the `useFetchOrders` hook in the `App` component to fetch and handle orders data.

The new folder structure now looks like this:

src/
|-- components/
| |-- orders/
| | |-- OrderRow.js
| | |-- OrderTable.js
| |-- App.js
|-- hooks/
| |-- useFetchOrders.js
|-- utils/
| |-- dateUtils.js
|-- index.js

These changes improve code modularity and maintainability by separating concerns into their respective folders. The custom hook `useFetchOrders` handles data fetching, and the `App` component uses this hook to manage the loading and error states for order data.

The code is now more organized, making it easier to understand and maintain.

#### Changes Made

1. **Pluralization of Model Name:**
   - Added the following code to `config/initializers/inflections.rb` to handle the pluralization of the model name:
     ```ruby
     ActiveSupport::Inflector.inflections(:en) do |inflect|
       inflect.irregular 'cookie', 'cookies'
     end
     ```

2. **Model Modifications:**
   - Updated the `Oven` model to have a `has_many :cookies` association. This change allows an oven to hold multiple cookies simultaneously, enabling us to place multiple cookies into the oven together.

3. **Quantity Field in Cookie Form:**
   - Added a virtual attribute `quantity` field to the cookie form. Now, when preparing cookies for baking, the user can specify the quantity of cookies they want to bake with the same fillings.

4. **Cookie Baking:**
   - Created the `Cookies/CreateService` to support baking multiple cookies with the same fillings. The service can now create the specified number of cookies in the oven, all with the same fillings. Custom validation ensures the quantity is greater than zero.

5. **Oven Show Component:**
   - Created a view component called `Oven::ShowComponent` to display the oven's information, including the number of cookies in the oven and the fillings of the first cookie (if any). The component enhances the organization and reusability of the view logic.

6. **View Components for All Views and Form:**
   - Created view components for all oven-related views, including the index, show, and form views. The use of view components improves the maintainability and modularity of the view code.

7. **Testing:**
   - Implemented thorough testing for the `Cookies/CreateService` to ensure it correctly handles the creation of multiple cookies with the same fillings.
   - Wrote unit tests for the  view components to validate the information displayed in the views.
   - All tests are written using RSpec and follow best practices for testing Rails applications.
 
## Issue 3 and Issue 4: Enhancing Cookie Baking Process and Real-time Oven Updates

This pull request addresses two issues:
- Issue 3: Cookies are now actually cooked in the bakery application.
- Issue 4: As a bakery owner, the oven page now updates automatically when cookies are ready.

### Changes Made

1. Added a new virtual attribute for cooking time:
   - A virtual attribute "cooking_time" has been introduced in the form.
   - When creating cookies, the baking time is now set using the "cooking_time" attribute.

2. Implementing Stimulus Controller for Real-time Updates:
   - Utilized the Stimulus Controller to achieve dynamic updates.
   - Displayed a loader and a timer until the baking time elapses.
   - Implemented polling to periodically check the cookies' status.

3. Button to Retrieve Cookies:
   - After the baking time is completed, a "Retrieve Cookies" button appears.

4. Achieving Real-time Updates:
   - Leveraged Turbo Streams to achieve real-time updates.
   - The oven page now automatically reflects changes in cookie status.

### Technical Details

- Introduced a virtual attribute "cooking_time" to accurately set baking time.
- Implemented a Stimulus Controller to manage real-time updates:
   - Displayed a loader and countdown timer until the baking time elapses.
   - Utilized polling to monitor the cookies' status.
   - Displayed the "Retrieve Cookies" button once baking is completed.
- Integrated Turbo Streams to achieve real-time updates on the oven page.
These enhancements provide an improved user experience, ensuring that cookies are accurately baked and the oven page updates in real-time, enhancing efficiency and convenience for bakery owners.
### Issue 5: Add Loading Indicator for Order Listing

- Added a loading indicator that appears while waiting for data from the API.
- Improved user experience by providing visual feedback during data retrieval.

### Issue 6: Sorting Order Listing Table

- Implemented sorting functionality for the orders in ascending (asc) and descending (desc) order.
- Users can now easily organize and view orders based on their preferences.

### Code Refactoring and Component Testing

- Refactored code to improve readability and maintainability.
- Thoroughly tested the OrdersTable and OrdersPage components to ensure functionality and reliability.

These enhancements enhance the user experience by introducing a loading indicator for data retrieval and enabling efficient sorting of order listings. The codebase has been refactored for better organization, and comprehensive testing has been performed to ensure the stability and correctness of the OrdersTable and OrdersPage components.

### Description

This pull request addresses Issue #7 by implementing the ability to mark orders as fulfilled using the `fulfilled` attribute in the orders database column. It introduces a new "Fulfill Order" action, allowing users to mark orders as fulfilled, and ensures that the changes are properly saved to the database.

### Changes Made

- Added a "Fulfill Order" action to the order management interface.
- Utilized the `fulfilled` attribute in the orders database column to manage order fulfillment status.
- Implemented necessary backend logic to update the fulfillment status in the database.
- Created corresponding frontend components and UI elements to support order fulfillment.
- Added unit tests to validate the correctness of the new functionality and ensure proper handling of order fulfillment actions.

### Additional Changes

In addition to the primary implementation, the following enhancements have been made:

- Improved UI/UX for order management to clearly indicate fulfillment status.
- Added loading indicators to provide visual feedback during the fulfillment process.
- Enhanced error handling for scenarios where order fulfillment encounters issues.
- Ensured that order details accurately display the current fulfillment status.

### Testing

Thorough testing has been conducted to validate the new functionality:

- Unit tests have been included to cover the implementation of the "Fulfill Order" action.
- Integration tests have been executed to verify seamless interaction between frontend and backend components.
- Edge cases and potential errors have been meticulously tested to ensure the system's robustness.

This pull request presents a substantial enhancement to the order management system, resulting in an improved user experience and precise tracking of order fulfillment status.

Your review and feedback are greatly appreciated.

Thank you.

### Update:

In this update:

- Replaced the script-based test execution in `package.json` with a Jest configuration file and a setup file in the root directory for more organized test management.

- Introduced a testing helper for testing Stimulus controllers, streamlining the testing process and ensuring consistent controller behavior.

- Added a new test for the `CookiesController` in Stimulus to verify its functionality and behavior.

- Restored the `Pagination` component to the codebase and conducted debugging to ensure proper integration with fulfillment actions when used in conjunction with paginations.

- Eliminated the usage of data-test-id attributes from the `OrderRaw` component, improving test readability and simplifying maintenance. The corresponding test for this component was also updated accordingly.

- Addressed React update warnings by wrapping asynchronous operations with the `act` function, ensuring that state updates are handled correctly during testing.

- Integrated Jest tests into GitHub Actions for automated testing and continuous integration.

### New Update:

- Refactored controllers and services for improved validation and real-time data display. Modified filling process to utilize icons and added Stimulus controllers for buttons and form reset. Updated tests. Known bug: Retrieving cookies after timer expiry requires two clicks - first for the "Retrieve" button and second for the "Oven is empty" message. Recognized antipattern: Using inline scripts for Tailwind CSS.

**Changes Made:**
- Controller and service code has been refactored to enhance validation and optimize real-time data rendering.
- Modified the way cookies are filled by incorporating icon usage, alongside the introduction of Stimulus controllers for buttons and form resetting.
- Test suite has been updated and improved to ensure robust functionality.
 -Added a GitHub Actions workflow to check and validate Tailwind CSS usage, ensuring adherence to best practices.

**Bug Notice:**
- A bug has been identified where two clicks are needed to retrieve cookies after the timer has expired and to display the "Oven is empty" message.

**Antipattern:**
- It has been recognized that using inline scripts for Tailwind CSS is not a recommended approach.

Thank you for your attention to these updates. We appreciate your ongoing collaboration and support.

Best regards,
Dejan

### CODE REVIEW

File: app/src/orders/components/OrderRow.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      storeFulfilledStatus(id, true);
+    } catch (error) {
+      console.error('Error fulfilling order:', error);
+      setIsFulfilling(false);
+    }
+  };
+
+  return (
+    <tr>
+      <td data-testid="order-id">{id}</td>
```
Comment (joelmoss) [inline app/src/orders/components/OrderRow.jsx:29]: test data/attributes should not be in non-test code.

**Comment by DeVuDeveloper** [comment]
Hi Joe,

Thank you for your prompt response and for reviewing the code. I had some
errors in the tests, which I added for debugging purposes. I've made
changes to both the component and the test, and I now have an update for
the pull request. I have some free time now, and I want to see how the Gem
Proscenium will collaborate with Tailwind CSS and Vite. After that, I'll
work on styling to make it resemble the Digital Oven design more closely.

Could you please take a look at the update? I'll also send you the update:
<https://github.com/DeVuDeveloper/test-app/pull/1>







*Replaced the script-based test execution in package.json with a Jest
configuration file and a setup file in the root directory for more
organized test management.Introduced a testing helper for testing Stimulus
controllers, streamlining the testing process and ensuring consistent
controller behavior.Added a new test for the CookiesController in Stimulus
to verify its functionality and behavior.Restored the Pagination component
to the codebase and conducted debugging to ensure proper integration with
fulfillment actions when used in conjunction with paginations.Eliminated
the usage of data-test-id attributes from the OrderRaw component, improving
test readability and simplifying maintenance. The corresponding test for
this component was also updated accordingly.Addressed React update warnings
by wrapping asynchronous operations with the act function, ensuring that
state updates are handled correctly during testing.Integrated Jest tests
into GitHub Actions.*

*Once again, thank you for your time.*

Best regards,
Dejan

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            ],
+            "parser": "@angular-eslint/template-parser"
+        }
+    ],
+    "rules": {
+        "tailwindcss/classnames-order": "warn",
+        "tailwindcss/no-custom-classname": "warn",
+        "tailwindcss/no-contradicting-classname": "error"
+    }
+}
```
Comment (muhammad-ur-rehman) [inline .eslintrc]: add empty line at the EOF

File: .github/workflows/linters.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      - name: Set up Node.js environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: npm install for Jest
+        run: npm install
+
+      - name: Run Jest tests
+        run: npm test -- --ci --coverage
```
Comment (muhammad-ur-rehman) [inline .github/workflows/linters.yml]: empty line EOF.
Please look at all of the files

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (muhammad-ur-rehman) [inline .vscode/settings.json]: why we need this file?

File: README.md [hunk 5]
```diff
@@ -1,127 +1,42 @@
...
+# Digital Bakery
 
-DigiBakery is an artisanal digital bakery, crafting the finest digital cookies in the world.
+## Screenshots
 
-We don't mass produce our cookies in faceless factories. Instead, We bake cookies to order, one at a time.
+<div style="display: flex; justify-content: space-between;">
+  <img src="app/assets/images/bakery1.png" alt="bakery 1" width="30%">
+  <img src="app/assets/images/bakery2.png" alt="bakery 2" width="30%">
+  <img src="app/assets/images/bakery3.png" alt="bakery 3" width="30%">
```
Comment (muhammad-ur-rehman) [inline README.md]: dont need to increase code size for the README file images are not required in readme until they are very important if it is important use any other free/paid source to place pictures there

File: __tests__/stimulus/cookies_controler.test.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    })
+  );
+
+  await screen.findByRole('button', { name: 'Remaining time: 0 seconds' });
+
+  expect(getHTML()).toMatchSnapshot();
+});
+
+
+
```
Comment (muhammad-ur-rehman) [inline __tests__/stimulus/cookies_controler.test.js:31]: empty spaces

File: app/components/ovens/cookie_info_component.html.erb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<%= turbo_frame_tag dom_id(@oven, "cookie_info") do %>
+
+  <div class="cookie-info flex flex-col justify-between text-center">
+    <% if @oven.cookies.any? { |cookie| cookie.cooked_at.present? } %>
+      <div>
+        <p class="text-orange-500">
+          <%= pluralize(@oven.cookies.size, "cookie") %> with <%= @oven.first_cookie_fillings || "no fillings" %>
```
Comment (muhammad-ur-rehman) [inline app/components/ovens/cookie_info_component.html.erb:7]: have you handled `@oven.cookies` (n+1) in the code base?

File: app/controllers/application_controller.rb [hunk 8]
```diff
@@ -1,3 +1,13 @@
...
   protect_from_forgery with: :exception
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:email, :password, :password_confirmation)
+    end
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:password, :password_confirmation, :current_password, :balance)
+    end
```
Comment (muhammad-ur-rehman) [inline app/controllers/application_controller.rb:11]: make these private i think we dont need them available in all controllers

File: app/controllers/cookies_controller.rb [hunk 9]
```diff
@@ -1,24 +1,84 @@
...
+              format.html { redirect_to ovens_path, alert: "Insufficient funds." }
+              format.turbo_stream { flash.now[:alert] = "Insufficient funds." }
+            end
+          end
+        else
+          @cookie = @oven.cookies.build(cookie_params)
+          flash.now[:alert] = validation_service.validate || create_service.error_message
+          respond_to do |format|
+            format.html { render :new, alert: flash.now[:alert] }
+            format.turbo_stream { flash.now[:alert] = flash.now[:alert] }
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb:35]: complexity of this method is very high reduce it

File: app/controllers/cookies_controller.rb [hunk 10]
```diff
@@ -1,24 +1,84 @@
...
+        create_service = Cookies::CreateService.new(current_user, @oven, cookie_params)
+        validation_service = Cookies::ValidationService.new(cookie_params)
+
+        if validation_service.validate.nil? && create_service.call
+          if current_user.balance >= paid_amount
+            current_user.subtract_from_balance(paid_amount)
+
+            respond_to do |format|
+              format.html { redirect_to ovens_path, notice: "Cookies were successfully created." }
+              format.turbo_stream.append("cookies", partial: "cookies/created_cookie")
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb:22]: too many render lines there should be only 1 render for html, 1 for tubo etc.

File: app/controllers/cookies_controller.rb [hunk 11]
```diff
@@ -1,24 +1,84 @@
...
+            respond_to do |format|
+              format.html { redirect_to ovens_path, alert: "Insufficient funds." }
+              format.turbo_stream { flash.now[:alert] = "Insufficient funds." }
+            end
+          end
+        else
+          @cookie = @oven.cookies.build(cookie_params)
+          flash.now[:alert] = validation_service.validate || create_service.error_message
+          respond_to do |format|
+            format.html { render :new, alert: flash.now[:alert] }
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb:34]: this method is doing magic?
you have to break this into smaller method follow SRP, DRY

File: app/controllers/cookies_controller.rb [hunk 12]
```diff
@@ -1,24 +1,84 @@
...
     end
   end
 
-  def create
-    @oven = current_user.ovens.find_by!(id: params[:oven_id])
-    @cookie = @oven.create_cookie!(cookie_params)
-    redirect_to oven_path(@oven)
+  def claim_change
+    paid_amount = params[:paid_amount].to_f
+    total_price = params[:total_price].to_f
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb]: always permit params before using

File: app/controllers/cookies_controller.rb [hunk 13]
```diff
@@ -1,24 +1,84 @@
...
+        format.turbo_stream { flash.now[:alert] = "Cannot create cookies when the oven is offline." }
+      end
     end
   end
 
-  def create
-    @oven = current_user.ovens.find_by!(id: params[:oven_id])
-    @cookie = @oven.create_cookie!(cookie_params)
-    redirect_to oven_path(@oven)
+  def claim_change
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb:47]: if this method is not linked with any route make it private

File: app/controllers/cookies_controller.rb [hunk 14]
```diff
@@ -1,24 +1,84 @@
...
+    paid_amount = params[:paid_amount].to_f
+    total_price = params[:total_price].to_f
+
+    change_to_claim = paid_amount - total_price
+
+    if change_to_claim.positive?
+      current_user.add_to_balance(change_to_claim)
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb:57]: what if this fails it will still return success

File: app/controllers/cookies_controller.rb [hunk 15]
```diff
@@ -1,24 +1,84 @@
...
+      format.turbo_stream { flash.now[:alert] = "Cookies are already in the oven!" }
+    end
+  end
+
+  def cookies_created_successfully
+    respond_to do |format|
+      format.html { redirect_to ovens_path, notice: "Cookies were successfully created." }
+      format.turbo_stream { flash.now[:notice] = "Cookies were successfully created." }
+    end
+  end
```
Comment (muhammad-ur-rehman) [inline app/controllers/cookies_controller.rb:82]: you need to slim the controller make use of concern or hellper or service what ever make sense

File: app/controllers/ovens_controller.rb [hunk 16]
```diff
@@ -1,17 +1,57 @@
...
+  def respond_to_oven_empty
+    respond_to do |format|
+      format.html { redirect_to oven_path(@oven) }
+      format.turbo_stream
+    end
+  end
+
+  def oven_status_json
+    {ready: @oven.first_cookie_ready?, time_left: @oven.time_left_for_first_cookie}
   end
```
Comment (muhammad-ur-rehman) [inline app/controllers/ovens_controller.rb:58]: code reusability + silm all the controllers

File: app/controllers/users_controller.rb [hunk 17]
```diff
@@ -0,0 +1,20 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!, only: [:show, :update, :add_balance]
```
Comment (muhammad-ur-rehman) [inline app/controllers/users_controller.rb:2]: this controller only have 1 method and you are authenticating for 3 methods

File: app/javascript/controllers/application.js [hunk 18]
```diff
@@ -0,0 +1,8 @@
+import { Application } from "@hotwired/stimulus"
+
+const application = Application.start()
+
+application.debug = false
+window.Stimulus   = application
+
+export { application }
```
Comment (muhammad-ur-rehman) [inline app/javascript/controllers/application.js:8]: extra line EOF

File: app/javascript/controllers/cookies_controller.js [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+    this.startPolling();
+  }
+
+  startPolling() {
+    this.pollingInterval = setInterval(() => {
+      this.fetchOvenStatus();
+    }, 5000);
+  }
+
+
```
Comment (muhammad-ur-rehman) [inline app/javascript/controllers/cookies_controller.js:17]: setup linter for these also

File: app/javascript/entrypoints/application.js [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+// Turbo.start()
+//
+// import ActiveStorage from '@rails/activestorage'
+// ActiveStorage.start()
+//
+// // Import all channels.
+// const channels = import.meta.globEager('./**/*_channel.js')
+
+// Example: Import a stylesheet in app/frontend/index.css
+// import '~/index.css'
```
Comment (muhammad-ur-rehman) [inline app/javascript/entrypoints/application.js:28]: remove unnecessary comments

File: app/javascript/entrypoints/application.js [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+console.log('Vite ⚡️ Rails')
+
+// If using a TypeScript entrypoint file:
+//     <%= vite_typescript_tag 'application' %>
+//
+// If you want to use .jsx or .tsx, add the extension:
+//     <%= vite_javascript_tag 'application.jsx' %>
+
+console.log('Visit the guide for more information: ', 'https://vite-ruby.netlify.app/guide/rails')
+
```
Comment (muhammad-ur-rehman) [inline app/javascript/entrypoints/application.js:15]: you are printing this on console for production as well?

File: app/services/cookies/create_service.rb [hunk 22]
```diff
@@ -0,0 +1,31 @@
+class Cookies::CreateService < ApplicationService
+  def initialize(user, oven, params)
+    @user = user
+    @oven = oven
+    @params = params
```
Comment (muhammad-ur-rehman) [inline app/services/cookies/create_service.rb:5]: how you know what variables you need in params

File: app/services/cookies/create_service.rb [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+    num_cookies_to_bake = @params[:quantity].to_i
+    cooking_time = @params[:cooking_time].to_i
+    price_per_cookie = @params[:price].to_f
+
+    @total_price = num_cookies_to_bake * price_per_cookie
+
+    num_cookies_to_bake.times do
+      cookie = @oven.cookies.build(@params)
+      cookie.save
+      cookie.update(cooked_at: Time.now + cooking_time)
```
Comment (muhammad-ur-rehman) [inline app/services/cookies/create_service.rb:24]: what if this breaks?

File: app/src/orders/App.jsx [hunk 24]
```diff
@@ -0,0 +1,16 @@
...
+      <OrdersPage />
+    </div>
+  );
+};
+
+export default App;
+
+
+
+
```
Comment (muhammad-ur-rehman) [inline app/src/orders/App.jsx:16]: linting issues

File: app/src/orders/components/OrderRow.jsx [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+  );
+};
+
+export default OrderRow;
+
+
+
+
+
+
```
Comment (muhammad-ur-rehman) [inline app/src/orders/components/OrderRow.jsx:58]: linting

File: app/src/orders/hooks/UseFetchOrders.jsx [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+  return { orders, loading, fetchOrders };
+};
+
+export default useFetchOrders;
+
+
+
+
+
+
```
Comment (muhammad-ur-rehman) [inline app/src/orders/hooks/UseFetchOrders.jsx:31]: linting

File: app/views/layouts/_navigation_links.html.erb [hunk 27]
```diff
@@ -6,7 +6,7 @@
     <%= link_to 'Sign out', destroy_user_session_path, method: 'delete', class: 'nav-link' %>
   </li>
   <li class="nav-item">
-    <%= link_to 'Order listing', orders_path, class: 'nav-link' %>
+    <%= link_to 'Order listing', orders_path, class: 'nav-link', data: {turbo: false} %>
```
Comment (muhammad-ur-rehman) [inline app/views/layouts/_navigation_links.html.erb:9]: one question why you need the turbo fase?

File: app/views/layouts/application.css [hunk 28]
```diff
@@ -1,5 +1,63 @@
...
+.flash__message {
+  font-size: 0.875rem;
+  animation: appear-then-fade 4s both;
+  border-radius: 999px;
+  margin-top: -10px;
+  margin-right: 100px;
+}
+
+.flash__message-notice {
+  color: hsl(120, 96%, 53%);
```
Comment (muhammad-ur-rehman) [inline app/views/layouts/application.css:31]: make use of variables in scss

File: app/views/ovens/_toggle.html.erb [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+        <button type="submit" class="mt-4">
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 96 96" class="svg-icon-red" style="width: 1.5em; height: 1.5em;">
+            <path d="M48 91.17a39.24 39.24 0 0 1-21.51-72 4 4 0 1 1 4.39 6.69 31.23 31.23 0 1 0 34.23 0 4 4 0 1 1 4.39-6.69 39.24 39.24 0 0 1-21.5 72Zm4-50V8.83a4 4 0 1 0-8 0v32.33a4 4 0 0 0 8 0Z" data-name="Artboard 30"></path>
+          </svg>
+        </button>
+      <% end %>
+
+
+
+
```
Comment (muhammad-ur-rehman) [inline app/views/ovens/_toggle.html.erb:13]: linting issues in all of the HTML files

**Review by muhammad-ur-rehman — CHANGES_REQUESTED** [review]

File: app/views/layouts/_navigation_links.html.erb (hunk 27 above)
Comment (DeVuDeveloper) [reply app/views/layouts/_navigation_links.html.erb:9]: It is React, so I don't need turbo.
